lldb/source/Core/Progress.cpp

```diff
@@ -82,20 +94,37 @@ ProgressManager &ProgressManager::Instance() {
   return *g_progress_manager;
 }
 
-void ProgressManager::Increment(std::string title) {
+void ProgressManager::Increment(Progress::ProgressData progress_data) {
```

Comment:
This should take a const ref to avoid the copy `const Progress::ProgressData& progress_data`. If we'd unconditionally store it in the map I'd say the copy is fine as you'd be able to move it into the map, but I'm assuming most of the time we're only using the title to do the lookup.

Reply:
Yeah, the title is used for map queries but the data itself isn't stored in the map, it's passed on to `ProgressManager::ReportProgress`.  If we change it to a const ref (`const Progress::ProgressData& progress_data`) here then we probably also want to do this for `Decrement` since it uses the map in a similar way.